server/game/cards/01_SOR/units/RukhThrawnsAssassin.ts

```diff
@@ -0,0 +1,28 @@
+import AbilityHelper from '../../../AbilityHelper';
+import { NonLeaderUnitCard } from '../../../core/card/NonLeaderUnitCard';
+
+
+export default class RukhThrawnsAssassin extends NonLeaderUnitCard {
+    protected override getImplementationId() {
+        return {
+            id: '3280523224',
+            internalName: 'rukh#thrawns-assassin',
+        };
+    }
+
+    public override setupCardAbilities() {
+        this.addTriggeredAbility({
+            title: 'Defeat unit being attacked',
+            when: {
+                onDamageDealt: (event, context) =>
```

Comment:
Oh this is a good callout! I was just thinking of this same issue recently, so I very recently added something that handles this in case of that exact problem. Basically now it is the case that CardTargetSystem (the parent class for any GameSystem that targets cards for effect) will return `canAffect = false` if a card is marked as defeated

https://github.com/SWU-Karabast/forceteki/blob/12647136235b6aeb31a978336ad903f82e69dc6b/server/game/core/gameSystem/CardTargetSystem.ts#L130